src/prefect/results.py

```diff
@@ -247,6 +263,20 @@ async def update_for_task(self: Self, task: "Task") -> Self:
             update["result_storage"] = await get_default_result_storage()
         return self.model_copy(update=update)
 
+    @staticmethod
+    def generate_default_holder() -> str:
```

Comment:
i didn't quite follow with this method was necessary - is it mainly to aid with testing or something else?

Reply:
Each lock has a `holder` associated with it and the goal with this method is to make providing a `holder` optional, but make it so that actors across machines, processes, and threads are identifiable. The alternative would be to make `holder` required for each actor. That's also doable. In most cases, the transaction would be the one creating the `holder`.

Reply:
ok i gotcha - i think it makes sense 👍